packages/entities/entities-plugins/src/components/PluginSelect.vue

```diff
@@ -0,0 +1,497 @@
+    type: Function as PropType<() => boolean | Promise<boolean>>,
+    required: false,
+    default: async () => true,
+  },
+  /**
+   * @param {boolean} noRouteChange if true, let consuming component handle event when clicking on a plugin
+   * Used in conjunction with `@plugin-clicked` event
+   */
+  noRouteChange: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+    * @param {boolean} onlyAvailablePlugins checks kong config plugins.available_on_server and if
+    * onlyAvailablePlugins = true, then it will not show plugins from PluginMeta that are outside
+    * of the available_on_server array.
+    */
+  onlyAvailablePlugins: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Plugins that should not be displayed
+   */
+  ignoredPlugins: {
+    type: Array as PropType<String[]>,
+    default: () => [],
+  },
+  /**
+   * Plugins that should be disabled and their disabled messages.
+   * ex. {
+   *  'acl': 'ACL is not supported for this entity type',
+   * }
+   */
+  disabledPlugins: {
+    type: Object as PropType<DisabledPlugin>,
+    default: () => ({}),
+  },
+  /**
+   * Number of plugins to always have visible (never will be collapsed)
+   */
+  pluginsPerRow: {
+    type: Number,
+    default: 4,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'loading', isLoading: boolean): void,
+  (e: 'plugin-clicked', plugin: PluginType): void,
+}>()
+
+const route = useRoute()
+const { i18n: { t } } = composables.useI18n()
+const { pluginMetaData } = composables.usePluginMetaData()
+const { getMessageFromError } = useErrors()
+const { sortAlpha, objectsAreEqual } = useHelpers()
+
+const filter = ref('')
+const isLoading = ref(true)
+const hasError = ref(false)
+const fetchErrorMessage = ref('')
+const availablePlugins = ref<string[]>([])
+const pluginsList = ref<PluginCardList>({})
+const existingEntityPlugins = ref<string[]>([])
+
+const { axiosInstance } = useAxios({
+  headers: props.config?.requestHeaders,
+})
+
+const filteredPlugins = computed((): PluginCardList => {
+  if (!pluginsList.value) {
+    return {}
+  }
+
+  const query = filter.value.toLowerCase()
+  const results = JSON.parse(JSON.stringify(pluginsList.value))
+
+  for (const type in pluginsList.value) {
+    const matches = pluginsList.value[type as keyof PluginCardList]?.filter((plugin: PluginType) => plugin.name.toLowerCase().includes(query) || plugin.group.toLowerCase().includes(query)) || []
+
+    if (!matches.length) {
+      delete results[type]
+    } else {
+      results[type] = matches
+    }
+  }
+
+  return results
+})
+
+const hasFilteredResults = computed((): boolean => {
+  return Object.keys(filteredPlugins.value).length > 0
+})
+
+const noSearchResults = computed((): boolean => {
+  return (Object.keys(pluginsList.value).length > 0 && !hasFilteredResults.value)
+})
+
+const tabs = props.config.app === 'konnect'
+  ? [
+    {
+      hash: '#kong',
+      title: t('plugins.select.tabs.kong.title'),
+    },
+    {
+      hash: '#custom',
```

Comment:
question: should these be i18n strings that get sanitized to alpha and `-` only?

Reply:
I'm not sure I follow, these are the actual tab hashes?

Reply:
Ignore this; I was going down a road when I reviewed but the implementation ended up being diff.